contracts/near/eth2-utility/src/consensus.rs

```diff
@@ -163,29 +239,41 @@ pub fn get_participant_pubkeys(
     result
 }
 
-pub fn convert_branch(branch: &[H256]) -> Vec<ethereum_types::H256> {
-    branch.iter().map(|x| x.0).collect()
+/// Verify a proof that `leaf` exists at `index` in a Merkle tree rooted at `root`.
+///
+/// The `branch` argument is the main component of the proof: it should be a list of internal
+/// node hashes such that the root can be reconstructed (in bottom-up order).
+pub fn verify_merkle_proof(
+    leaf: H256,
+    branch: &[H256],
+    depth: usize,
+    index: usize,
+    root: H256,
+) -> bool {
+    if branch.len() == depth {
+        merkle_root_from_branch(leaf, branch, depth, index) == root
+    } else {
+        false
+    }
 }
 
-pub fn validate_beacon_block_header_update(header_update: &HeaderUpdate) -> bool {
-    let branch = convert_branch(&header_update.execution_hash_branch);
-    if branch.len() != EXECUTION_PROOF_SIZE {
-        return false;
+/// Compute a root hash from a leaf and a Merkle proof.
+pub fn merkle_root_from_branch(leaf: H256, branch: &[H256], depth: usize, index: usize) -> H256 {
```

Comment:
I guess we should remove similar function from `eth2near/eth_rpc_client/src/executon_block_proof.rs`

Reply:
Yes, but there are different types of conflict issues, so it will be better to solve them in separate PR.